CRM/Core/CommunityMessages.php

```diff
@@ -130,11 +130,12 @@ public function getDocument() {
    *   parsed JSON
    */
   public function fetchDocument() {
-    list($status, $json) = $this->client->get($this->getRenderedUrl());
-    if ($status != CRM_Utils_HttpClient::STATUS_OK || empty($json)) {
+    $httpget = CRM_Utils_Http::get($this->getRenderedUrl());
```

Comment:
You should write this with a unit test from the start - it will change how you write it - in this case you will want to have a way to pass the client in - here is a test line eg. https://github.com/eileenmcnaughton/omnimail-silverpop/blob/master/test/tests/SilverpopTest.php#L30

Later on you can see it does $client->request(

https://github.com/eileenmcnaughton/silverpop-php-connector/blob/master/src/SilverpopConnector/SilverpopXmlConnector.php#L179

I'm not saying to follow exactly that pattern, only that you should try writing a unit test as you write it because it will change it. 

If you are wanting to work on this in advance of Dec when 5.3 goes you could probably create an extension that overwrites community messages & get ti going in there